app/cdash/include/filterdataFunctions.php

```diff
@@ -983,7 +983,7 @@ function parse_filter_from_request($field_var, $compare_var, $value_var,
     $fieldinfo = explode('/', $fieldinfo, 2);
     $field = $fieldinfo[0];
     $compare = htmlspecialchars(pdo_real_escape_string($_REQUEST[$compare_var]));
-    $value = htmlspecialchars(pdo_real_escape_string($_REQUEST[$value_var]));
+    $value = pdo_real_escape_string($_REQUEST[$value_var]);
```

Comment:
Note that this change applies to _all_ filter values, not just test names. I think this is safe since we're still calling `pdo_real_escape_string()` (and our tests still pass) but I wanted to highlight this change since it's potentially more wide ranging.